avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep

```
@@ -0,0 +1,1030 @@
+              delegations: [
+                {
+                  name: 'Microsoft.App.environments'
+                  properties: {
+                    serviceName: 'Microsoft.App/environments'
+                  }
+                }
+              ]
+            }
+          ]
+        : [],
+      contains(computeTypes, 'azure-container-app') && privateNetworking && selfHostedConfig.selfHostedType == 'azuredevops'
+        ? [
+            {
+              name: networkingConfiguration.?containerAppDeploymentScriptSubnetName ?? 'app-deployment-script-subnet'
+              addressPrefix: networkingConfiguration.?containerAppDeploymentScriptSubnetPrefix ?? '10.0.3.0/29'
+            }
+          ]
+        : [],
+      []
+    )
+  }
+}
+module appEnvironment 'br/public:avm/res/app/managed-environment:0.6.2' = if (contains(
+  computeTypes,
+  'azure-container-app'
+)) {
+  name: 'appEnv-${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'appEnv${namingPrefix}${uniqueString(resourceGroup().id)}'
+    logAnalyticsWorkspaceResourceId: logAnalyticsWokrspace.outputs.resourceId
+    location: location
+    infrastructureSubnetId: networkingConfiguration.networkType == 'createNew'
+      ? filter(
+          newVnet.outputs.subnetResourceIds,
+          subnetId => contains(subnetId, networkingConfiguration.?containerAppSubnetName ?? 'app-subnet')
+        )[0]
+      : networkingConfiguration.networkType == 'useExisting'
+          ? '${networkingConfiguration.virtualNetworkResourceId}/subnets/${networkingConfiguration.computeNetworking.containerAppSubnetName}'
+          : null
+    zoneRedundant: true
+    internal: privateNetworking ? true : false
+    workloadProfiles: [
+      {
+        name: 'consumption'
+        workloadProfileType: 'consumption'
+      }
+    ]
+  }
+}
+
+module natGatewayPublicIp 'br/public:avm/res/network/public-ip-address:0.5.1' = if (empty(networkingConfiguration.?natGatewayResourceId ?? '') && empty(networkingConfiguration.?natGatewayPublicIpAddressResourceId ?? '') && networkingConfiguration.networkType == 'createNew' && privateNetworking) {
+  name: 'natGatewayPublicIp-${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'natGatewayPublicIp-${uniqueString(resourceGroup().id)}'
+    skuName: 'Standard'
+    publicIPAddressVersion: 'IPv4'
+    publicIPAllocationMethod: 'Static'
+  }
+}
+
+module natGateway 'br/public:avm/res/network/nat-gateway:1.1.0' = if (privateNetworking && empty(networkingConfiguration.?natGatewayResourceId ?? '') && networkingConfiguration.networkType == 'createNew') {
+  name: 'natGateway-${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'natGateway-${namingPrefix}-${uniqueString(resourceGroup().id)}'
+    zone: 0
+    location: location
+    publicIpResourceIds: empty(networkingConfiguration.?natGatewayPublicIpAddressResourceId ?? '')
+      ? [natGatewayPublicIp.outputs.resourceId]
+      : [networkingConfiguration.?natGatewayPublicIpAddressResourceId ?? '']
```

If I'm not mistaken, you should be able to simplify this and similar statements to:

```suggestion
    publicIpResourceIds: [
      networkingConfiguration.?natGatewayPublicIpAddressResourceId ?? natGatewayPublicIp.outputs.resourceId
    ]
```

---

Resolved